docs/data/material/guides/shadow-dom/shadow-dom.md

```diff
@@ -0,0 +1,64 @@
+# Shadow DOM
+
+<p class="description"><a href="https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM" target="_blank" rel="noopener nofollow">Shadow DOM</a> can be helpful to isolate part of the app from unwanted global styles.</p>
+
+## Steps
+
+### 1. Styles
+
+In order for styles to work in Shadow DOM, they need to be applied inside of the Shadow DOM:
+
+```tsx
+const container = document.querySelector('#root');
+const shadowContainer = container.attachShadow({ mode: 'open' });
+const emotionRoot = document.createElement('style');
+const shadowRootElement = document.createElement('div');
+shadowContainer.appendChild(emotionRoot);
+shadowContainer.appendChild(shadowRootElement);
+
+const cache = createCache({
+  key: 'css',
+  prepend: true,
+  container: emotionRoot,
+});
+
+ReactDOM.createRoot(shadowRootElement).render(
+  <CacheProvider value={cache}>
+    <App />
+  </CacheProvider>,
+);
+```
+
+### 2. Theme
+
+MUI components like `Menu`, `Dialog`, `Popover` and others use [React Portal](https://reactjs.org/docs/portals.html) to render a new "subtree" in a container outside of current DOM hierarchy.
+By default, this container is `document.body`. But since the styles are applied only inside of the Shadow DOM, we need to render portals inside the Shadow DOM container as well:
```

Comment:
```suggestion
By default, this container is `document.body`. 
But since the styles are applied only inside of the shadow DOM, we need to render portals inside the shadow DOM container as well:
```

Reply:
Newline doesn't make a difference in markdown.
Did you mean to split it into two paragraphs? An additional empty line is needed then

Reply:
I like to see each sentence on a new line to make it simpler to comment on individual sentences (rather than paragraphs). Sorry I should have explained that! It's a pattern that I think I saw @oliviertassinari enforce when I started contributing to the docs. Might be worth adding as a prettier rule if it's something we really want to standardize.

Reply:
Ah, so you meant just a line break in markdown, right?